redis/asyncio/client.py

```diff
@@ -1043,7 +1043,7 @@ async def run(
                 await self.get_message(
                     ignore_subscribe_messages=True, timeout=poll_timeout
                 )
-            except asyncio.CancelledError:
+            except (asyncio.CancelledError, GeneratorExit):
```

Comment:
I've never actually reproduced this @kozzztik . Can you (or community @kristjanvalur maybe) weigh in here?

I don't want to catch a random exception I have yet to reproduce.

Reply:
The thinking is correct. GeneratorExit generally should not be handled, any more than CancelledError. However I'd prefer to change this so that the next 'except' clause catches Exception, rather than BaseException. BaseException should almost never be caught. CancelledError still needs its own reraise handler for 3.7 compatibility.